packages/runtime/container-runtime/src/summaryFormat.ts

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SummaryType } from "@fluidframework/protocol-definitions";
+import { channelsTreeName, ISummaryTreeWithStats } from "@fluidframework/runtime-definitions";
+
+interface IFluidDataStoreAttributes1 {
+    pkg: string;
+    readonly snapshotFormatVersion?: "0.1";
+    /**
+     * This tells whether a data store is root. Root data stores are never collected.
+     * Non-root data stores may be collected if they are not used. If this is not present, default it to
+     * true. This will ensure that older data stores are incorrectly collected.
+     */
+    readonly isRootDataStore?: boolean;
+}
+interface IFluidDataStoreAttributes2 extends IFluidDataStoreAttributes1 {
+    readonly snapshotFormatVersion?: undefined;
+    readonly summaryFormatVersion: 2;
+    /** True if channels are not isolated in .channels subtrees, otherwise isolated. */
+    readonly disableIsolatedChannels?: true;
+}
+/**
+ * Added IFluidDataStoreAttributes similar to IChannelAttributes which will tell the attributes of a
+ * store like the package, snapshotFormatVersion to take different decisions based on a particular
+ * snapshotFormatVersion.
+ */
+export type ReadFluidDataStoreAttributes = IFluidDataStoreAttributes1 | IFluidDataStoreAttributes2;
+export type WriteFluidDataStoreAttributes = IFluidDataStoreAttributes2;
+
+export function getAttributesFormatVersion(attributes: ReadFluidDataStoreAttributes): number {
+    if ("summaryFormatVersion" in attributes) {
```

Comment:
Is `if (attributes.summaryFormatVersion)` a more definitive check here?

Reply:
Do you mean in case it's undefined? Typescript won't allow that check by default, but I can change it to also verify that it is not undefined.

Reply:
My question was more around the pattern - `property in object` vs `object.property !== undefined`. I am not sure if it makes a difference or if one is better than the other.